### PR TITLE
Add methods for remaining VisualElement properties

### LINKED ISSUE #142: [Proposal] Add methods for all VisualElement Properties
# Add methods for all VisualElement Properties

* [x] Proposed
* [x] Prototype
* [x] Implementation: [PR raised](https://github.com/CommunityToolkit/Maui.Markup/pull/141)
  * [x] iOS Support
  * [x] Android Support
  * [x] macOS Support
  * [x] Windows Support
* [x] Unit Tests: at least one for each new method following the conventions for other methods in the class
* [x] Sample: Some uses of new methods added to the Markup sample app
* [x] Documentation: [PR raised in docs repo](https://github.com/MicrosoftDocs/CommunityToolkit/pull/166)

## Link to Discussion

https://github.com/CommunityToolkit/Maui.Markup/discussions/140

## Summary
[summary]: #summary

As part of ensuring that Maui.Markup can provide a fluent interface for all standard controls, it will be a good starting point for having fluent methods for all properties of `VisualElement` as this is the base class to a lot of common controls.

## Motivation
[motivation]: #motivation

Fill in some common gaps in the fluent methods for defining a control instance.
Avoid needing to use the clunky method of combining an object initializer and chained fluent methods.

## Detailed Design
[design]: #detailed-design

Fill in the gaps of extension methods for setting the remaining properties of `VisualElement` that do not have them.

The properties are:

- AnchorX
- AnchorY
- Background
- BackgroundColor
- Clip
- FlowDirection
- Opacity
- InputTransparent
- IsEnabled
- IsVisible
- RotationX
- RotationY
- Rotation
- ScaleX
- ScaleY
- TranslationX
- TranslationY
- ZIndex

These should all be simple extension methods that wrap setting the property and then returning the element.

The implementations will be of the style:

```csharp
public static TVisualElement AnchorX<TVisualElement >(this TVisualElement element, double anchorX) where TVisualElement : VisualElement
{
	element.AnchorX = anchorX;
	return element;
}
```

In addition to simple methods for the above properties, I also think it would be useful to add the following helper methods:

A method `Anchor()` that takes two values that can be applied to `AnchorX` and `AnchorY` with a single call.
A method `Scale()` that takes two values that can be applied to `ScaleX` and `ScaleY` with a single call.

## Usage Syntax
[usage]: #usage-syntax

With this change, it will be possible to write

```cs
new Label().Text("hello").Opacity(0.5).BackgroundColor(Colors.Red);
```

rather than having to write 

```cs
new Label{ Opacity = 0.5, BackgroundColor = Colors.Red }.Text("hello")
```

## Drawbacks
[drawbacks]: #drawbacks

It makes the assembly slightly larger by adding methods where some of these might be rarely used. 
It feels an appropriate trade-off for completeness though.

## Alternatives
[alternatives]: #alternatives

No real alternatives other than not doing this.
It just makes it easier to provide a more consistent way of writing code when defining a UI in C#.

## Unresolved Questions
[unresolved]: #unresolved-questions

n/a

### PR DESCRIPTION
**Pull request by mrlacey**
For discussion #140

 ### Description of Change ###

 

Add methods for properties of VisualElement that do not yet have them.

Also updates the Sample app to use these new methods (where appropriate).
Also fixes the image not displaying on the sample app settings page.

 ### Linked Issues ###
 

 - Fixes #142

 ### PR Checklist ###
 
 - [x] Has a linked Issue, and the Issue has been `approved`(bug) or `Championed` (feature/proposal)
 - [x] Has tests (if omitted, state reason in description)
 - [x] Has samples (if omitted, state reason in description)
 - [x] Rebased on top of `main` at time of PR
 - [x] Changes adhere to [coding standard](https://github.com/CommunityToolkit/Maui.Markup/blob/main/CONTRIBUTING.md#contributing-code---best-practices)
 - [x] Documentation created or updated: https://github.com/MicrosoftDocs/CommunityToolkit/pull/166 

 ### Additional information ###

### CODE REVIEW

File: src/CommunityToolkit.Maui.Markup/VisualElementExtensions.cs
Comment (pictos) [inline src/CommunityToolkit.Maui.Markup/VisualElementExtensions.cs:202]: Can we change the generic type here and in others to `TVisualElement`? The reason is that MAUI has a type named `Element` and this can be confusing, mainly if we want to add some extensions methods that target `Element`

**Review by pictos — CHANGES_REQUESTED** [review]
Hey @mrlacey thanks for your PR! I've one request, let me know what do you think

**Comment by brminnick** [comment]
Thanks @mrlacey! 

Before we can merge this PR, there's a couple procedural steps we first need to do:
1. Submit a [New Feature Proposal](https://github.com/CommunityToolkit/Maui.Markup/issues/new?assignees=&labels=new%2C+proposal&template=open-a-new-feature-proposal.md&title=%5BProposal%5D+)
2. Get Approval Vote from Core Maintainer Team

When you get a chance, go ahead and open a [Proposal](https://github.com/CommunityToolkit/Maui.Markup/issues/new?assignees=&labels=new%2C+proposal&template=open-a-new-feature-proposal.md&title=%5BProposal%5D+). Then, I'll bring this Visual Element Proposal to a vote at our next [monthly standup](https://www.youtube.com/watch?v=kf1sPqhkbRA).

We have some more info in the README about the new feature process. It's a bit of a pain, but it ensures that every new feature we add is super high quality (which this one looks to be!).
https://github.com/CommunityToolkit/Maui.Markup#submitting-a-new-feature
[](https://github.com/CommunityToolkit/Maui.Markup#submitting-a-new-feature)

File: src/CommunityToolkit.Maui.Markup/VisualElementExtensions.cs
Comment (mrlacey) [reply src/CommunityToolkit.Maui.Markup/VisualElementExtensions.cs]: Changed as requested. For the existing methods in the class too. 
For reference there already are extension methods for `Element` and they use the type name `TView`.
It appears there is some general inconsistency around how typenames should be determined. If all are to be reviewed I'd rather do that as a separate task and not part of the PR.

**Review by brminnick — APPROVED** [review]
Thanks so much @mrlacey!!

**Pull request merged**
